web/modules/custom/rdf_etl/tests/src/Kernel/AdmsValidationStepTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Drupal\Tests\rdf_etl\Kernel;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\rdf_entity\RdfEntityGraphStoreTrait;
+use Drupal\Tests\rdf_entity\Traits\RdfDatabaseConnectionTrait;
+use EasyRdf\Graph;
+
+/**
+ * Tests the 'adms_validation' process step plugin.
+ *
+ * @group rdf_etl
+ */
+class AdmsValidationStepTest extends KernelTestBase {
+
+  use RdfDatabaseConnectionTrait;
+  use RdfEntityGraphStoreTrait;
+
+  /**
+   * The testing sink graph.
+   *
+   * @var string
+   */
+  const TEST_GRAPH = 'http://example.com/graph/test/sink';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'adms_validator',
+    'rdf_entity',
+    'rdf_etl',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->setUpSparql();
+    // Make sure that the SPARQL testing graph is empty before running any test.
+    $this->sparql->query("CLEAR GRAPH <" . static::TEST_GRAPH . ">;");
+  }
+
+  /**
+   * Tests the ADMS validation step.
+   *
+   * @param string $rdf_file
+   *   The RDF file to be tested.
+   * @param bool $valid
+   *   Expectancy: The file is a valid ADMS v2 file.
+   *
+   * @throws \Exception
+   *   If the plugin is invalid.
+   *
+   * @dataProvider providerTestAdmsValidationStepPlugin
+   */
+  public function testAdmsValidationStepPlugin(string $rdf_file, bool $valid): void {
+    /** @var \Drupal\rdf_etl\Plugin\EtlProcessStepManager $manager */
+    $manager = \Drupal::service('plugin.manager.etl_process_step');
+    $data = ['sink_graph' => static::TEST_GRAPH];
+    $plugin = $manager->createInstance('adms_validation', $data);
+
+    $graph = new Graph();
+    $graph->parseFile(__DIR__ . "/../../fixtures/$rdf_file");
```

Comment:
Let's let drupal handle the path. We can use
```
$filepath = drupal_get_path('module', 'rdf_etl') . "/tests/fixtures/{$rdf_file}";
$graph->parseFile($filepath);
```
instead.

Reply:
When you are referring a file within the same extension (module, theme, profile) you should use the path relative to `__DIR__` instead of `drupal_get_path()`. Learn here why https://www.drupal.org/project/drupal/issues/2351919